### PR TITLE
[docs] Update Logstash output documentation

### LINKED ISSUE #1631: Missing instructions about different indices when using logstash output
Following the format the data is sent when using the elasticsearch output directly, as seen here:

https://github.com/elastic/apm-server/blob/6.5/apm-server.yml#L409-L431

The docs for the logstash output are not correct: https://www.elastic.co/guide/en/apm/server/current/logstash-output.html

There is no message saying that it should forward the data based on the `process.event` value.

Here is a `logstash.conf` that takes that into consideration, and works for other beats components:
```
input {
  beats {
    port => 5044
  }
}

filter {
  if [@metadata][beat] == "apm-server" {
    if [processor][event] == "sourcemap" {
      mutate { add_field => { "[@metadata][index]" => "apm-%{[@metadata][version]}-sourcemap" } }
    } else {
      mutate { add_field => { "[@metadata][index]" => "apm-%{[@metadata][version]}-%{[processor][event]}-%{+yyyy.MM.dd}" } }
    }
  } else {
    mutate { add_field => { "[@metadata][index]" => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}" } }
  }
}

output {
  elasticsearch {
    hosts => ["http://localhost:9200"]
    index => "%{[@metadata][index]}"
  }
}
```

### PR DESCRIPTION
**Pull request by bmorelli25**
This PR updates the output config documentation for Logstash. Closes #1631. These changes will also need to be made in elastic/beats.

**A preview of the following changes to `logstash-output.html` is available [here](http://apm-server_2691.docs-preview.app.elstc.co/guide/en/apm/server/master/logstash-output.html).**

* Replaces "The Beat" with "APM Server"
* Updates `@metadata` information
* Adds two sample logstash configs for APM users

Here is the logstash config I got to work with APM. It'd be great if someone with more logstash experience could take a look and recommend any improvements.

```conf
input {
    beats {
        port => 5044
    }
}

filter {
    if [@metadata][beat] == "apm" {
        if [processor][event] == "sourcemap" {
            mutate {
                add_field => { "[@metadata][index]" => "%{[@metadata][beat]}-%{[@metadata][version]}-%{[processor][event]}" } <1>
            }
        } else {
            mutate {
                add_field => { "[@metadata][index]" => "%{[@metadata][beat]}-%{[@metadata][version]}-%{[processor][event]}-%{+yyyy.MM.dd}" } <2>
            }
        }
    }
}

output {
    elasticsearch {
        hosts => ["http://localhost:9200"]
        index => "%{[@metadata][index]}"
    }
}
```

In addition, the following configs work with manual ILM. Is one preferred over the others?

```conf
input {
    beats {
        port => 5044
    }
}

output {
    elasticsearch {
        hosts => ["http://localhost:9200"]
        index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{[processor][event]}"
    }
}
```
vs

```conf
input {
    beats {
        port => 5044
    }
}

filter {
    if [@metadata][beat] == "apm" {
        mutate {
            add_field => { "[@metadata][index]" => "%{[@metadata][beat]}-%{[@metadata][version]}-%{[processor][event]}" }
        }
    }
}

output {
    elasticsearch {
        hosts => ["http://localhost:9200"]
        index => "%{[@metadata][index]}"
    }
}
```

### CODE REVIEW

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 1]
```diff
@@ -739,21 +739,41 @@ use in Logstash for indexing and filtering:
...
-WARNING: The `@metadata.type` field, added by the Logstash output, is
-deprecated, hardcoded to `doc`, and will be removed in {beatname_uc} 7.0.
+ifdef::apm-server[]
+["source","json",subs="attributes"]
+------------------------------------------------------------------------------
+{
+    ...
+    "@metadata": { <1>
+      "beat": "{beat_default_index_prefix}", <2>
+      "pipeline":"apm", <3>
```
Comment (bmorelli25) [inline docs/copied-from-beats/outputconfig.asciidoc]: I thought automatic pipeline registration requires `output.elasticsearch` to be `true`? Why am I seeing this here?

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 2]
```diff
@@ -739,21 +739,41 @@ use in Logstash for indexing and filtering:
...
+    }
+}
+------------------------------------------------------------------------------
+<1> {beatname_uc} uses the `@metadata` field to send metadata to Logstash. See the
+{logstash-ref}/event-dependent-configuration.html#metadata[Logstash documentation]
+for more about the `@metadata` field.
+<2> The default is {beat_default_index_prefix}. To change this value, set the
+<<logstash-index,`index`>> option in the {beatname_uc} config file.
+<3> Why am I seeing pipeline here?
+<4> The IP address of APM Server .
```
Comment (bmorelli25) [inline docs/copied-from-beats/outputconfig.asciidoc]: ?

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 1 above)
Comment (simitt) [reply docs/copied-from-beats/outputconfig.asciidoc]: The registration of the pipeline does require ES output, as it registers the pipeline on startup/setup. Applying pipelines on events during ingestion is also possible for logstash output.

File: docs/copied-from-beats/outputconfig.asciidoc [hunk 3]
```diff
@@ -739,21 +739,41 @@ use in Logstash for indexing and filtering:
...
+    }
+}
+------------------------------------------------------------------------------
+<1> {beatname_uc} uses the `@metadata` field to send metadata to Logstash. See the
+{logstash-ref}/event-dependent-configuration.html#metadata[Logstash documentation]
+for more about the `@metadata` field.
+<2> The default is {beat_default_index_prefix}. To change this value, set the
+<<logstash-index,`index`>> option in the {beatname_uc} config file.
+<3> Why am I seeing pipeline here?
+<4> The IP address of APM Server.
```
Comment (simitt) [inline docs/copied-from-beats/outputconfig.asciidoc]: Why are you adding the IP address here?

**Comment by bmorelli25** [comment]
Hey @simitt. Thanks for the review and sorry this is confusing. It was late and I didn't do a good job of asking my questions. Here's what I'm really getting at:

Our current docs only mention four metadata fields that you can use for Logstash filtering and indexing: 
Screen Shot 2019-09-10 at 10 01 17 AM

However, when I look at the metadata object in `7.3`, this is what I see:
```
   {  
      "beat":"apm",
      "pipeline":"apm",
      "ip_address":"127.0.0.1",
      "type":"_doc",
      "version":"7.3.1"
   }
```

So I have three questions:

* `type` has been deprecated and hardcoded to `_doc` for a long time. Does it make sense to still document it?
* What is `ip_address` and should we document it?
* Pipeline it sounds like it makes sense to document.

**Comment by simitt** [comment]
> type has been deprecated and hardcoded to _doc for a long time. Does it make sense to still document it?

I agree not to document it any more.

> What is ip_address and should we document it?

I am honestly not aware of where the IP address is coming from.. 

> Pipeline it sounds like it makes sense to document.

Fully agree, it is set to `apm` by default. See [logstash#use-ingest-pipelines](https://www.elastic.co/guide/en/logstash/current/use-ingest-pipelines.html) for reference of logstash config. If the users want to use another pipeline, they should directly configure it in logstash config, as we don't have a way of configuring it in the `apm-server.yml` for output different than ES.

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: I removed this field

File: docs/copied-from-beats/outputconfig.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/copied-from-beats/outputconfig.asciidoc]: Removed. Not sure why this is showing up.

**Comment by bmorelli25** [comment]
Wonderful, thank you! I've updated the PR

**Pull request merged**